src/t_set.c

```diff
@@ -778,8 +779,12 @@ void spopWithCountCommand(client *c) {
             alsoPropagate(c->db->id,propargv,3,PROPAGATE_AOF|PROPAGATE_REPL);
             decrRefCount(objele);
 
-            lp = lpDelete(lp, p, &p);
+            /* Store pointer for later deletion and move to next. */
+            ps[i] = p;
+            p = lpNext(lp, p);
+            index++;
         }
+        lp = lpBatchDelete(lp, ps, count);
```

Comment:
forgot to free `ps` (also in "case 3)

Reply:
Thx, yeah I found out from the failed CI.